src/main/java/com/synopsys/integration/alert/database/api/user/UserAccessor.java

```diff
@@ -0,0 +1,153 @@
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.database.api.user;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.synopsys.integration.alert.database.relation.UserRoleRelation;
+import com.synopsys.integration.alert.database.relation.repository.UserRoleRepository;
+import com.synopsys.integration.alert.database.user.RoleEntity;
+import com.synopsys.integration.alert.database.user.RoleRepository;
+import com.synopsys.integration.alert.database.user.UserEntity;
+import com.synopsys.integration.alert.database.user.UserRepository;
+
+@Component
+public class UserAccessor {
+    public static final String DEFAULT_ADMIN_USER = "sysadmin";
+    private final UserRepository userRepository;
+    private final RoleRepository roleRepository;
+    private final UserRoleRepository userRoleRepository;
+    private final PasswordEncoder defaultPasswordEncoder;
+
+    @Autowired
+    public UserAccessor(final UserRepository userRepository, final RoleRepository roleRepository, final UserRoleRepository userRoleRepository, final PasswordEncoder defaultPasswordEncoder) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.userRoleRepository = userRoleRepository;
+        this.defaultPasswordEncoder = defaultPasswordEncoder;
+    }
+
+    @Transactional
+    public List<UserModel> getUsers() {
+        final List<UserEntity> userList = userRepository.findAll();
+        return userList.stream().map(this::createModel).collect(Collectors.toList());
+    }
+
+    @Transactional
+    public Optional<UserModel> getUser(final String username) {
+        final Optional<UserModel> userResult;
+        final Optional<UserEntity> entity = userRepository.findByUserName(username);
```

Comment:
This might be a good place to use Optional.map(). You could do something like: 
`return userRepository.findByUserName(username).map(this::createModel);`
(I have been putting the chained methods on their own lines like streams for readability, but markdown doesn't let me do it here for some reason).